src/transfer-manager.ts

```diff
@@ -664,8 +696,12 @@ export class TransferManager {
     let partNumber = 1;
     let promises: Promise<void>[] = [];
     try {
+      if (options.abortExisting && options.uploadId) {
```

Comment:
How does this option work? If I set abortExisting to true, and have a upload ID, it will cancel the the upload? Shouldn't it only do this if it fails?

Reply:
oh, looks like you can abort an upload using `uploadFileInChunks` when you set abortExisting=True and uploadId is set; based on 

>  [abortExisting] boolean to indicate if an in progress upload should be aborted. uploadID must also be supplied in order to abort the upload.

I guess if this delete in context of this method; the upload id would be known and isn't required like this.

Reply:
@danielduhh correct the user would have to pass back the `uploadId` and set the `abortExisting` option. The way it works currently is if there is a failure the return has the `uploadId` and mapping of the completed parts that were uploaded.